src/components/FedSignIn/FedSignIn.tsx

```diff
@@ -46,9 +48,20 @@ const FedSignIn = ({ setUserId }: FedSignInProps) => {
     try {
       const result = await getRedirectResult(auth);
       if (result?.user) {
+        const userDocRef = doc(db, "users", result.user.uid);
+
+      await setDoc(userDocRef, {
+        firstName: result?.user.displayName?.split(" ")[0],
+        lastName: result?.user.displayName?.split(" ")[1],
+        email: result?.user.email,
+        UUID: auth?.currentUser?.uid,
+        events: [],
+      });
```

Comment:
can you make this block into a function as you are using it in the register page too. You could also wrap it in a try catch block as we probably need to catch the error if it doesn't update the database. The function should be written in the firebaseSnapshots utility file